backend/s3/s3.go

```diff
@@ -1959,6 +1967,24 @@ func (f *Fs) getMemoryPool(size int64) *pool.Pool {
 	)
 }
 
+// PublicLink generates a public link to the remote path (usually readable by anyone)
+func (f *Fs) PublicLink(ctx context.Context, remote string) (link string, err error) {
+	if ok, err := fs.FileExists(ctx, f, remote); !ok {
```

Comment:
The way to do this in a backend is to call `f.NewObject(ctx, remote)` and if you get any errors just return them.